### ISSUE TITLE
[Screen Reader - MakeCode - Microbit Menu Controls] Screen reader is reading incorrect info for the search field value after performing delete operation.

### ISSUE THREAD

**Issue by v-mesala** [Microsoft/pxt#2627]
**User Experience:**
If Screen reader is reading as end of line deselect even after erasing the text in the search field then users with visual disabilities might be confuse to understand the control.

**Test Environment**
OS: RS2 Version 1703(OS Build 15063.483)
Platform: Edge.
Screen reader: Narrator

**Repro Steps**
1-Navigate to https://makecode.microbit.org/acc
2-Navigate to the Micro bit section element and select Code control given on it.
3-Navigate to the micro bit menu controls and edit in search field.
4-Erase the text in the search field and navigate to search.
5-Navigate using Shift + Tab to the search editable field and verify the issue.

**Actual Result**
1.Screen reader is reading as end of line deselect even after erasing the text in the search field
2.Visually, after backward Navigation Search editable field looks as some empty text selected.

**Expected Result**
Visually search place holder should visible and screen reader should read as search editable field.

**MAS Reference**
https://microsoft.sharepoint.com/teams/msenable/Documents/LCA/MAS%20-%20July%201,%202017%20Release/MAS%201.3.1%20%E2%80%93%20Info%20and%20Relationships%20.docx?d=w54f28d1fa2d14dcd84e15c9b87e8aba4&csf=1

**Suggested fix:**
1- ARIA roles/attributes must be used wherever possible to support AT’s in a better way. 
 
Roles: 
1.1) Landmarks Roles: All the sections should be aligned with respective landmark role 
e.g. 
`<div id="header" role="banner">A banner image and introductory title</div> `
`<div id="sitelookup" role="search">....</div> `
 
1.2) Headings Roles : All the heading elements should be aligned with respective role or role and aria-level 
e.g. 
`<div role="heading">Global News items</div> `
... a list of global news with editorial comment... 
 
`<div role="heading" aria-level="7">Jonagold/div> `
`<p>Jonagold is a cross between the Golden Delicious and Jonathan varieties...</p> `
 
1.3) Roles associated with Labels: sections/regions of the webpage should be aligned with label’s  
e.g. 
`<div role="application" aria-labelledby="p123" aria-describedby="info"> `
`<h1 id="p123">Calendar<h1> `
`<p id="info"> `
This calendar shows the game schedule for the Boston Red Sox. 
`</p>`
 
Reference: 
https://www.w3.org/TR/WCAG20-TECHS/ARIA11.html  
https://www.w3.org/TR/WCAG20-TECHS/ARIA12.html  
https://www.w3.org/TR/WCAG20-TECHS/ARIA13.html  
 
2 - All the elements should be programmatically aligned correctly  
 
-      Paragraph elements should be tagged with `<p>` tag 
-      Link elements should have `<a href>` 
-      Buttons should have `<a>` or  `<a role=”button”> `
-      Only interactive elements should be aligned with <a> tag, static elements on the page should not be made interactive by using `<a>` tag  
Reference: 
https://www.w3.org/WAI/WCAG20/quickref/?showtechniques=111%2C211%2C131#content-structure-separation-programmatic  
 
3 – Correct Usage of ARIA attributes in web pages 
 
3.1) Required owned elements 
If any element(parent) having role=’menu’(parent role) then it should have at least one element(child) which will have role=’menu item’(child role) 
e.g.       
 `<div role="menu" title=”File Menu”> `
  ` <div role="menuitem" title=”File-Help”>` 
 3.2) Correct scope 
If any element(child) having role=’radio’(child role) then it should be under an element(parent) which has role=’radio group’(parent role) 
 e.g.  
    ` <div role="radiogroup" title=”File Menu”> `
            ` <div role="radio" title=”File-Help”> `
 
3.3) State & Properties of ARIA attributes should be valid 
ARIA attributes should use only the allowed or supported values as per WCAG 2.0 guidelines 
e.g. 
`<div role="radiogroup" aria-labelledby=”idvalue”>  `
allowed value of aria-labelledby attribute is ID Reference,  
It will be an issue if any other string/numeric is used for the attribute 
 
3.4) Valid Roles 
Only valid non-abstract roles must be used 
e.g. 
`<div role="button" >` is valid role 
`<div role="datepicker”>` is invalid role 
  
3.5) Unsupported ARIA attributes 
Only supported ARIA attributes should be used as per the respective role/tag of the element  
e.g. 
`<div role="radio" aria-required=”true”> aria-required is not supported for role=radio >`
`<div role="listbox" aria-expanded=”true”> aria-expanded is not supported for role=radio > `

Reference: 
https://www.w3.org/TR/wai-aria/roles#mustContain 
https://www.w3.org/TR/wai-aria/roles#scope  
https://www.w3.org/TR/wai-aria/states_and_properties#propcharacteristic_value  
https://www.w3.org/TR/wai-aria/roles#roles_categorization 
https://www.w3.org/TR/wai-aria/roles#supportedState  

**Refer Attachment:**
[MAS 1.3.1_Makecode_Search editable field.zip](https://github.com/Microsoft/pxt/files/1193627/MAS.1.3.1_Makecode_Search.editable.field.zip)

**Comment by veler** [comment]
This is part of Blocky, a third-part component.

**Comment by microsoftsam** [comment]
@veler Although this is inside the Blockly editor, we add this searchbox in blocklyloader.ts in pxt. We should be able to fix this easily.

**Comment by veler** [comment]
Alright, I didn't know that. I assigned the issue to me and will investigate it.

**Comment by v-mesala** [comment]
Bug is verified on URL :https://makecode.microbit.org/acc ans bug is fixed Hence closed the bug and added the Closed tag.
